packages/fxa-content-server/app/scripts/views/pair/pairing-totp-mixin.js

```diff
@@ -9,22 +9,34 @@ export default function () {
     hasTotpEnabledOnAccount(profile) {
       return profile.authenticationMethods && profile.authenticationMethods.includes('otp');
     },
+    cancelPairingWithError(err) {
+      this.replaceCurrentPage('pair/failure', {
+        error: err,
+      });
+    },
     checkTotpStatus() {
       const account = this.getSignedInAccount();
 
       if (! account) {
-        this.replaceCurrentPage('pair/failure', {
-          error: AuthErrors.toError('UNKNOWN_ACCOUNT'),
-        });
+        return this.cancelPairingWithError(AuthErrors.toError('UNKNOWN_ACCOUNT'));
       }
 
       return account.accountProfile().then((profile) => {
-        // pairing is disabled for accounts with 2FA
-        if (this.hasTotpEnabledOnAccount(profile)) {
-          this.replaceCurrentPage('pair/failure', {
-            error: AuthErrors.toError('TOTP_PAIRING_NOT_SUPPORTED'),
+        if (this.model.get('totpComplete')) {
+          // once the user with TOTP successfully verifies we check the /exists endpoint
+          // it can throw on unverified sessions or tell us if TOTP doesn't exist
```

Comment:
@rfk here's what we got

Reply:
I find the logic here a little hard to follow - if `totpComplete` has been set, doesn't that mean that we already know the user has a TOTP token and they're good to go? Why do we need to `checkTotpTokenExists()` here? Are we worried that the previous page may have set `totpComplete` incorrectly? Is it just to double-check that the session is verified? (It should definitely be verified if we just came back from the TOTP page).